dask_ml/model_selection/_incremental.py

```diff
@@ -347,3 +394,558 @@ def fit(
+            y_test,
+            additional_calls=self._additional_calls,
+            fit_params=fit_params,
+            scorer=scorer,
+            random_state=self.random_state,
+        )
+        results = self._process_results(results)
+        history_results = self._get_history_results(results)
+        best_estimator, best_index = self._get_best(results, history_results)
+        best_estimator = yield best_estimator
+
+        # Clean up models we're hanging onto
+        ids = list(results.models)
+        for model_id in ids:
+            del results.models[model_id]
+
+        self.scorer_ = scorer
+        self.history_results_ = history_results
+        self.best_estimator_ = best_estimator
+        self.best_index_ = best_index
+        self.best_score_ = history_results[best_index]["score"]
+        self.best_params_ = history_results[best_index]["params"]
+        self.n_splits_ = 1
+        self.multimetric_ = False  # TODO: is this always true?
+        raise gen.Return(self)
+
+    def fit(self, X, y, **fit_params):
+        """Find the best parameters for a particular model.
+
+        Parameters
+        ----------
+        X, y : array-like
+        **fit_params
+            Additional partial fit keyword arguments for the estimator.
+        """
+        return default_client().sync(self._fit, X, y, **fit_params)
+
+    @if_delegate_has_method(delegate=("best_estimator_", "estimator"))
+    def predict(self, X, y=None):
+        self._check_is_fitted("predict")
+        return self.best_estimator_.predict(X)
+
+    @if_delegate_has_method(delegate=("best_estimator_", "estimator"))
+    def predict_proba(self, X):
+        self._check_is_fitted("predict_proba")
+        return self.best_estimator_.predict_proba(X)
+
+    @if_delegate_has_method(delegate=("best_estimator_", "estimator"))
+    def predict_log_proba(self, X):
+        self._check_is_fitted("predict_log_proba")
+        return self.best_estimator_.predict_log_proba(X)
+
+    @if_delegate_has_method(delegate=("best_estimator_", "estimator"))
+    def decision_function(self, X):
+        self._check_is_fitted("decision_function")
+        return self.best_estimator_.decision_function(X)
+
+    @if_delegate_has_method(delegate=("best_estimator_", "estimator"))
+    def transform(self, X):
+        self._check_is_fitted("transform")
+        return self.best_estimator_.transform(X)
+
+    @if_delegate_has_method(delegate=("best_estimator_", "estimator"))
+    def inverse_transform(self, Xt):
+        self._check_is_fitted("inverse_transform")
+        return self.best_estimator_.transform(Xt)
+
+    def score(self, X, y=None):
+        if self.scorer_ is None:
+            raise ValueError(
+                "No score function explicitly defined, "
+                "and the estimator doesn't provide one %s" % self.best_estimator_
+            )
+        return self.scorer_(self.best_estimator_, X, y)
+
+
+class RandomizedIncrementalSearch(BaseIncrementalSearch):
```

Comment:
Wasn't this going to be renamed to `StopOnPlateauSearch` (with other comment `StopOnPlateauSearchCV`)?

Reply:
I don't think that the term `StopOnPlateau` is easily recognizable by non-experts.

Reply:
I'm not super in-favor of `RandomizedIncrementalSearch` either though.  Given that this is now a subset of functionality with `ExponentialDecaySearch` (also not a great name) I think that we should consider merging them, choosing a more generic name, and providing a couple of keywords to choose between some common choices (exponential decay, patience, grid/random).